### PR TITLE
Location attachment draft & deserialize_as & check permission on attachment

### PR DESCRIPTION
**Pull request by bitfriend**
- Implement location attachment draft
- Replace `deserialize()` for specified msg event with `deserialize_as()`. Will not touch `deserialize()` that covers multiple cases.
- Remove unused import
- Check permission when sending attachment
- Simplify fn name

### CODE REVIEW

File: native/acter/src/api/attachments.rs [hunk 1]
```diff
@@ -361,6 +362,26 @@ impl AttachmentsManager {
...
 
+    pub fn location_attachment_draft(
+        &self,
+        body: String,
+        geo_uri: String,
+    ) -> Result<AttachmentDraft> {
+        if !self.is_joined() {
+            bail!("Can only attachment in joined rooms");
+        }
+        let mut builder = self.inner.draft_builder();
```
Comment (gtalha07) [inline native/acter/src/api/attachments.rs:373]: I think we would also want to add line for room member post permission check here too?

File: native/acter/src/api/uniffi_api.rs [hunk 2]
```diff
@@ -1,13 +1,12 @@
-use crate::{api::NotificationItem as ApiNotificationItem, login_with_token};
-use matrix_sdk::ruma::events::room::message::MessageType;
-use matrix_sdk::ruma::events::AnySyncMessageLikeEvent;
-use matrix_sdk::ruma::events::AnySyncTimelineEvent;
-use matrix_sdk::ruma::events::SyncMessageLikeEvent;
-
+use matrix_sdk::ruma::events::{
```
Comment (gtalha07) [inline native/acter/src/api/uniffi_api.rs:1]: Are these changes related to ```deserialize_as()``` change?

**Review by gtalha07 — APPROVED** [review]
LGTM!

File: native/acter/src/api/attachments.rs (hunk 1 above)
Comment (bitfriend) [reply native/acter/src/api/attachments.rs:361]: Fixed by https://github.com/acterglobal/a3/pull/1196/commits/dff6d53fbca074513d5a396cb3eb4bc1eb40f256
